java/flight/flight-integration-tests/src/test/java/org/apache/arrow/flight/integration/tests/IntegrationTest.java

```diff
@@ -114,6 +124,8 @@ void testScenario(String scenarioName) throws Exception {
           scenario.client(allocator, location, client);
         }
       }
+      exec.shutdown();
+      final boolean unused = exec.awaitTermination(3, TimeUnit.SECONDS);
```

Comment:
Sorry, do you mind documenting why we need the wait? 

@vibhatha can we link this to a new issue? I think this is possibly the source of other test flakiness that we've been unable to track down over the years

Reply:
Document as in add a comment in the code?

Reply:
added

Reply:
@lidavidm will create an issue.

Reply:
https://github.com/apache/arrow/issues/43948